### PR TITLE
GINConv

### PR DESCRIPTION
**Pull request by TristanBilot**
This PR introduces:

* GINConv
* Modifiable `aggr` in all layers
* Handle `edge_weights` by default in all MPNN layers
* `edge_weights` implem' for GCN

## Checklist

Put an `x` in the boxes that apply.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

### CODE REVIEW

File: mlx_graphs/nn/conv/gin_conv.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        array([[-0.536501, 0.154826, 0.745569, ..., 0.31547, -0.0962588, -0.108504],
+            [-0.415889, -0.0498145, 0.597379, ..., 0.194553, -0.251498, -0.207561],
+            [-0.119966, -0.0159533, 0.276559, ..., 0.0258303, -0.194533, -0.15515],
+            [-0.21477, -0.169684, 0.485867, ..., 0.0194768, -0.145761, -0.139433],
+            [-0.133289, -0.0279559, 0.358095, ..., -0.0443346, -0.11571, -0.114396]], dtype=float32)
+    """
+
+    def __init__(
+        self,
+        mlp: nn.Module,
```
Comment (francescofarina) [inline mlx_graphs/nn/conv/gin_conv.py:60]: can we rename it to `model` or something similar?

File: mlx_graphs/nn/conv/gin_conv.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        \mathbf{x}_i + \sum_{j \in \mathcal{N}(i)} \mathbf{x}_j \right)
+
+    where :math:`\mathbf{x}_i` and  :math:`\mathbf{h}_i` represent the input features
+    and output embeddings of node :math:`i`, respectively. :math:`\text{MLP}` denotes
+    a custom neural network provided by the user and :math:`\epsilon` is an epsilon
+    value either fixed or learned.
+
+    Args:
+        mlp: Callable :class:`mlx.core.nn.Module` applied on the final node embeddings
+        eps (optional): Initial value of the :math:`\epsilon` term. Default: ``0``
```
Comment (francescofarina) [inline mlx_graphs/nn/conv/gin_conv.py:24]: Please remove all the `(optional)` from the docstrings. Types are inferred from the signature and if a value is provided, like `(optional)`, the actual type is overwritten in the docs.

**Review by francescofarina — CHANGES_REQUESTED** [review]
Looking good - just two comments to address before merging.

File: mlx_graphs/nn/conv/gin_conv.py (hunk 1 above)
Comment (TristanBilot) [reply mlx_graphs/nn/conv/gin_conv.py:60]: `mlp` is how it's called in the original paper, `model`  is too general. Do you prefer `nn` or `neural_net`?

Comment (TristanBilot) [reply mlx_graphs/nn/conv/gin_conv.py:60]: Or even `encoder`

File: mlx_graphs/nn/conv/gin_conv.py (hunk 2 above)
Comment (TristanBilot) [reply mlx_graphs/nn/conv/gin_conv.py]: You're right

File: mlx_graphs/nn/conv/gin_conv.py (hunk 1 above)
Comment (TristanBilot) [reply mlx_graphs/nn/conv/gin_conv.py:60]: The MLP is included in the equation, so it's quite clear where it's used when looking at the docs of the layer

Comment (francescofarina) [reply mlx_graphs/nn/conv/gin_conv.py:60]: ok, let's leave it as it is

**Review by francescofarina — APPROVED** [review]

**Pull request merged**